modules/apps/frontend-theme/frontend-theme-classic/src/css/custom_properties/_custom_properties_set.scss

```diff
@@ -50,79 +50,130 @@ body {
 
 // Buttons
 
-.btn-link:not(.dropdown-item) {
-	@include button-cp(
-		var(--transparent),
-		var(--transparent),
-		var(--primary),
-		var(--transparent),
-		var(--transparent),
-		var(--primary-d1)
-	);
-}
+.btn:not(.btn-link) {
+	position: relative;
+	z-index: 1;
+
+	&::after {
+		background-color: black;
+		border-radius: clay-enable-rounded($btn-border-radius);
+		bottom: math-sign($btn-border-width);
+		content: '';
+		left: math-sign($btn-border-width);
+		opacity: 0;
+		position: absolute;
+		right: math-sign($btn-border-width);
+		top: math-sign($btn-border-width);
+		transition: opacity 150ms ease-in-out;
+		will-change: opacity;
+		z-index: -1;
+	}
 
-.btn-outline-light {
-	@include button-cp(
-		var(--transparent),
-		var(--white),
-		var(--white),
-		var(--white),
-		var(--white),
-		var(--dark)
-	);
-}
+	&:hover,
+	&:focus {
+		&::after {
+			opacity: 0.2;
+		}
+	}
 
-.btn-outline-primary {
-	@include button-cp(
-		var(--transparent),
-		var(--primary),
-		var(--primary),
-		var(--primary-d1),
-		var(--primary-d1),
-		var(--white)
-	);
-}
+	&:active,
+	&.active {
+		&::after {
+			opacity: 0.3;
+		}
+	}
 
-.btn-outline-secondary {
-	@include button-cp(
-		var(--transparent),
-		var(--gray-400),
-		var(--secondary),
-		var(--gray-200),
-		var(--transparent),
-		var(--dark)
-	);
+	&.disabled,
+	&:disabled {
+		&::after {
+			opacity: 0;
+		}
+	}
 }
 
-.btn-primary {
-	@include button-cp(
-		var(--primary),
-		var(--primary),
-		var(--white),
-		var(--primary-d1),
-		var(--primary-d1)
+.btn-link:not(.dropdown-item) {
```

Comment:
I'm not a big fan of the `:not(.selector)` pattern. It increases specificity by 1 class. This might cause some problems for older themes. These styles will win over someone who did:
###### _custom.scss
```
.btn-link { color: red; }
```
person upgrading needs to change selector to:
```
.btn-link:not(.dropdown-item) { color: red; }
```

Reply:
In this phase I think we can't remove the `not` but we'll definitely try once we move everything to Clay 🤞